### PR TITLE
csi: CSI Deployment documentation

### PR DESCRIPTION
**Pull request by lpabon**

### CODE REVIEW

File: scheduler/kubernetes/csi.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+## Install
+
+### Install using the Portworx spec generator
+When installing Portworx through [install.portworx.com](https://install.portworx.com)
+you can select CSI as the model to use for deployment.
+
+If you are using [curl to fetch the Portworx
+spec](https://docs.portworx.com/scheduler/kubernetes/px-k8s-spec-curl.html), you can add
+`csi=true` to the parameter list to include Stork specs in the generated file.
```
Comment (piyush-nimbalkar) [inline scheduler/kubernetes/csi.md]: *to include CSI specs

File: scheduler/kubernetes/csi.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+For any StorageClasses created, you will need to setup the value of `provisioner`
+to `com.openstorage.pxd`. Here is an example:
+
+```yaml
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: portworx
+provisioner: com.openstorage.pxd
+  repl: "3"
```
Comment (piyush-nimbalkar) [inline scheduler/kubernetes/csi.md]: indentation for provisioner
and parameters section

File: scheduler/kubernetes/csi.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+layout: page
+title: "Deploying Portworx with CSI support"
+keywords: csi, portworx, container, Kubernetes, storage, Docker, k8s, pv, persistent disk 
+sidebar: home_sidebar
+---
+
+* TOC
+{:toc}
+
+[CSI](https://kubernetes-csi.github.io/), or _Container Storage Interface_, is
```
Comment (piyush-nimbalkar) [inline scheduler/kubernetes/csi.md]: should we mention that CSI is in beta in K8s?

Comment (lpabon) [reply scheduler/kubernetes/csi.md]: Yeah, I'll add that.

File: scheduler/kubernetes/csi.md (hunk 2 above)
Comment (lpabon) [reply scheduler/kubernetes/csi.md]: provisioner indentation is correct, but you I missed the parameters section.

File: scheduler/kubernetes/csi.md (hunk 3 above)
Comment (lpabon) [reply scheduler/kubernetes/csi.md]: done

File: scheduler/kubernetes/csi.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+With CSI, Kubernetes gives storage drivers the opportunity to release on their
+schedule. This allows storage vendors to upgrade, update, and enhance their drivers
+without the need to update Kubernetes, maintaining a consistent, dependable,
+orchestration system.
+
+## Install
+
+### Install using the Portworx spec generator
+When installing Portworx through [install.portworx.com](https://install.portworx.com)
```
Comment (piyush-nimbalkar) [inline scheduler/kubernetes/csi.md]: Currently we don't expose the CSI option on the install.portworx.com page. It's hidden.
I think we can keep the docs same and just add the option on the install page. What do you think?

File: scheduler/kubernetes/csi.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  repl: "3"
+  snap_interval: "70"
+```
+
+Your PersistentVolumes and PersistentVolumeClaims will all work the same as before.
+
+## Upgrade
+
+Currently upgrades are _not_ supported. You will need to deploy using CSI onto
+an new Kubernetes cluster. The Kubernetes community is working very hard to make
```
Comment (harsh-px) [inline scheduler/kubernetes/csi.md]: grammar fix `You will need to deploy using CSI onto a new Kubernetes cluster`

File: scheduler/kubernetes/csi.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+```yaml
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: portworx
+provisioner: com.openstorage.pxd
+parameters:
+  repl: "3"
+  snap_interval: "70"
```
Comment (harsh-px) [inline scheduler/kubernetes/csi.md]: remove `snap_interval`. No need to take snaps be default.

**Pull request merged**